src/utils/JsonValidation.cpp

```
@@ -0,0 +1,19 @@
+#include "JsonValidation.h"
+
+#include "utils/Log.h"
+
+bool JsonValidation::containsKeys(nlohmann::json json_data, ValidationSettings settings) {
+    for (const auto& [key, value_type] : settings) {
+
+        if (!json_data.contains(key)) {
+            Log::log_print(DEBUG, std::format("Missing key {} in object", key).c_str());
```

this should probably throw an exception